pages/premium/konto/components/UserForm.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useState } from "react";
+import cn from "classnames";
+import BreakitApi from "@utils/breakitApi";
+import styles from "./styles/UserForm.module.scss";
+import { validEmail, validString } from "utils/validator";
+import Input from "components/atoms/Input";
+import Button from "components/atoms/Button";
+
+type UserDataType = {
+  id: number;
+  name: string;
+  email: string;
+  email_verified_at: Date;
+  password: string;
+  active: boolean;
+  firstname: string;
+  lastname: string;
+  jobtitle: string;
+  phone: string;
+  stripe_customer_id: string;
+  remember_token: string;
+  country: string;
+  vat_number: string;
+  represents_company: string;
+  last_login_at: Date;
+};
+
+type UserType = {
+  user: UserDataType;
+};
+
+const UserForm = ({ user }: UserType) => {
+  console.log(user);
+  const [email, setEmail] = useState(user?.email);
+  const [firstname, setFirstname] = useState(user?.firstname);
+  const [lastname, setLastname] = useState(user?.lastname);
+  const [clientErrors, setClientErrors] = useState({
+    emailError: false,
+    firstnameError: false,
+    lastnameError: false,
+  });
+
+  const updateUserData = () => {
+    BreakitApi.users
+      .update(user.id, user)
+      .then((response: any) => {
+        console.log("success updating user");
+        //setUser(response);
+      })
+      .catch((error: any) => {
+        console.log("Error updating user: ", error);
+      });
+  };
+
+  const handleChange = (e: any) => {
+    const { name, value } = e.target;
+    const cleanValue = name === "email" ? value.trimStart() : value;
+    if (name === "email") {
+      setEmail(cleanValue);
+    } else if (name === "name") {
+      setFirstname(cleanValue);
+    } else {
+      setLastname(cleanValue);
+    }
+  };
+  const handleSubmit = (e: any) => {
+    e.preventDefault();
+    console.log("Pressed zeh button");
+
+    let errors = {
+      emailError: false,
+      firstnameError: false,
+      lastnameError: false,
+    };
+    if (!validEmail(email)) {
+      errors.emailError = true;
+    }
+
+    if (!validString(firstname)) {
+      errors.firstnameError = true;
+    }
+
+    if (!validString(lastname)) {
+      errors.lastnameError = true;
+    }
```

Comment:
Vad tror du om det här? Då behöver värdena inte sättas om.
```suggestion
    const errors = {
      emailError: !validEmail(email),
      firstnameError: !validString(firstName),
      lastnameError: !validString(lastName),
    };
```

Reply:
Snyggt!